File: Easy/Horse-racing Duals/Java/HorseRacingDuals.java
import java.util.*;

class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);
        int numberOfHorses = in.nextInt();

        int[] horsesStrenghts = new int[numberOfHorses];

        for (int i = 0; i < numberOfHorses; i++)
            horsesStrenghts[i] = in.nextInt();

        Arrays.sort(horsesStrenghts);

        int minStrenghtDifference = horsesStrenghts[1] - horsesStrenghts[0];

        for (int i = 0; i < numberOfHorses - 1; i++)
            minStrenghtDifference = Math.min(minStrenghtDifference, (horsesStrenghts[i + 1] - horsesStrenghts[i]));

        System.out.println(minStrenghtDifference);
    }
}

File: Easy/Horse-racing Duals/JavaScript/Horse-racing Duals.js
numberOfHorses = parseInt(readline());

horsesStrenghts = [];

for (i = 0; i < numberOfHorses ; i++)
    horsesStrenghts[i] = parseInt(readline());

horsesStrenghts = horsesStrenghts.sort(function (a, b) {
    if (a > b) return 1;
    if (a < b) return -1;
})

minStrenghtDifference = horsesStrenghts[1] - horsesStrenghts[0];
for (i = 0; i < numberOfHorses - 1; i++)
    minStrenghtDifference = Math.min(minStrenghtDifference, horsesStrenghts[i + 1] - horsesStrenghts[i]);

print(minStrenghtDifference);